### PR TITLE
Public Links Data Service

### PR DESCRIPTION
**Pull request by refs**
## Disclaimer

This PR depends on https://github.com/cs3org/reva/pull/822, since it is a PR from a fork of mine, go modules replace will not work, and I will restrain from pushing a `go.mod` with a replace to a local path. If you would like to try this branch locally you would need to:

1. `git clone github.com/cs3org/reva`
2. `hub checkout https://github.com/cs3org/reva/pull/822`
3. on `ocis-reva` `go.mod` => `replace github.com/cs3org/reva => your/path/to/reva`
4. on `ocis`: `replace github.com/owncloud/ocis-reva => path/to/your/local/ocis-reva`
5. on `ocis`: `replace github.com/cs3org/reva => path/to/your/local/reva`
6. on `ocis`: `go run cmd/ocis/main.go server`

### CODE REVIEW

**Comment by refs** [comment]
This is a work in progress until https://github.com/cs3org/reva/pull/822 is merged, once merged, `go.mod` needs to be updated.

**Comment by ownclouders** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/command/storagepubliclinkdata.go  8
- pkg/flagset/storagepubliclinkdata.go  1
         

Clones added
============
- pkg/command/storagepubliclink.go  3
- pkg/flagset/storagepubliclink.go  5
- pkg/command/storagepubliclinkdata.go  14
- pkg/flagset/storagepubliclinkdata.go  22
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/ocis-reva/pullRequest?prid=5674853&bid=18579197)

**Comment by refs** [comment]
> - pkg/flagset/storagepubliclinkdata.go  22

💀

**Comment by PVince81** [comment]
@refs please set assignments, labels and project so that this becomes visible in the sprint.

once again it took me a while to realize that this is what I needed to check out from reviewing as there is no obvious link from https://github.com/cs3org/reva/pull/822

**Comment by PVince81** [comment]
@refs seems we're missing a PR to ocis as well to start that new data provider service ?

running `bin/ocis server` on the ocis repo master doesn't start that service

**Comment by refs** [comment]
> @refs seems we're missing a PR to ocis as well to start that new data provider service ?
> 
> running `bin/ocis server` on the ocis repo master doesn't start that service

true. I was considering deploying my dev binaries so that people can test them.

File: pkg/command/storagepubliclink.go [hunk 1]
```diff
@@ -82,10 +82,22 @@ func StoragePublicLink(cfg *config.Config) *cli.Command {
...
 						"services": map[string]interface{}{
 							"publicstorageprovider": map[string]interface{}{
-								"mount_path":   cfg.Reva.StoragePublicLink.MountPath,
-								"mount_id":     cfg.Reva.StoragePublicLink.MountID,
-								"driver_addr":  cfg.Reva.StoragePublicLink.PublicShareProviderAddr,
-								"gateway_addr": cfg.Reva.Gateway.URL,
+								"gateway_addr":       cfg.Reva.Gateway.URL,
+								"mount_path":         cfg.Reva.StoragePublicLink.MountPath,
+								"mount_id":           cfg.Reva.StoragePublicLink.MountID,
+								"driver_addr":        cfg.Reva.StoragePublicLink.PublicShareProviderAddr,
```
Comment (PVince81) [inline pkg/command/storagepubliclink.go:88]: rename setting to "public_share_provider_addr" and adjust also in reva ? "driver_addr" is too confusing

**Comment by PVince81** [comment]
Obsolete, we don't need a data service for public links as the public storage provider relays everything to the regular storage provider + regular data service, see https://github.com/cs3org/reva/pull/822#issuecomment-648227346

**Pull request closed without merging**